core/src/main/java/org/hibernate/ogm/dialect/spi/AssociationTypeContext.java

```diff
@@ -20,13 +20,20 @@
 public interface AssociationTypeContext {
 
 	/**
-	 * Provide access to the options set for an entity or property.
+	 * Provide access to the options set for the association.
 	 *
-	 * @return a context object providing access to the options effectively applying for a given entity or property.
+	 * @return a context object providing access to the options effectively applying for the association.
 	 */
 	OptionsContext getOptionsContext();
 
 	/**
+	 * Provide access to the options set for the entity owner of the association.
+	 *
+	 * @return a context object providing access to the options effectively applying for the owner entity.
+	 */
+	OptionsContext getOwnerEntityOptionsContext();
```

Comment:
> What I meant is, can we somehow generalize that, though?

That would be interesting as e.g. Infinispan also supports TTL, and I assume at least Ehcache would have it too.

But let's define the use case clearly first, and assign it a (low?) priority accordingly.

Let's also consider that both Ehcache and Infinispan can configure a TTL for the whole storage (the Cache, an of course support other smarter forms of eviction too), so supporting such a feature for Hibernate OGM is only interesting for use cases which require a per-instance override of the global eviction settings.